runtime/compiler/runtime/JITServerAOTCache.cpp

```diff
@@ -1430,16 +1590,240 @@ JITServerAOTCacheMap::~JITServerAOTCacheMap()
    }
 
 
+std::string
+JITServerAOTCacheMap::buildCacheFileName(const std::string &cacheDir, const std::string &cacheName)
+   {
+   std::string cacheFileName;
+   // If a directory is given, add it in the front of the cache name
+   if (!cacheDir.empty())
+      cacheFileName = cacheDir + "/";
+   return cacheFileName + "JITServerAOTCache." + cacheName + ".J" + std::to_string(JAVA_SPEC_VERSION);
+   }
+
+
+void
+JITServerAOTCacheMap::queueAOTCacheForSavingToFile(const std::string &cacheName)
+   {
+   OMR::CriticalSection cs(_monitor);
+   _cachesToSaveQueue.push_back(cacheName);
+   }
+
+
+bool
+JITServerAOTCacheMap::saveNextQueuedAOTCacheToFile()
```

Comment:
Out of curiosity, why isn't there a need for a `PersistentUnorderedSet<std::string> _cachesBeingSaved`, analogous to how for cache loads, there's both a queue to hold the caches to load as well as a set to prevent other threads from loading the same cache?

Reply:
For cache stores we can write the information in the cache itself (see `bool _saveOperationInProgress;`). 
For cache loads, the cache objects do not exist, so we need a separate data structure.

Reply:
Ah right makes sense.